frame/support/src/storage/mod.rs

```diff
@@ -44,71 +47,123 @@ pub mod types;
 pub mod unhashed;
 pub mod weak_bounded_vec;
 
-#[cfg(all(feature = "std", any(test, debug_assertions)))]
-mod debug_helper {
-	use std::cell::RefCell;
+mod transaction_level_tracker {
+	type Layer = u32;
+	const TRANSACTION_LEVEL_KEY: &'static [u8] = b":transaction_level:";
+	const TRANSACTIONAL_LIMIT: Layer = 255;
 
-	thread_local! {
-		static TRANSACTION_LEVEL: RefCell<u32> = RefCell::new(0);
+	pub fn get_transaction_level() -> Layer {
+		crate::storage::unhashed::get_or_default::<Layer>(TRANSACTION_LEVEL_KEY)
```

Comment:
can we use a global variable, well, for a non persistent global variable?

Reply:
You mean something like this? https://github.com/paritytech/substrate/compare/master...kiz-shawntabrizi-transactional-rework2

Reply:
yes

Reply:
I think environmental is very cumbersome in this case. Using storage is clearly not the right way to do it but it works for now. In this case we could just use atomics which compile into plain arithmetic for wasm.